README.md

```diff
@@ -1 +1,49 @@
-# java-jaxrs
\ No newline at end of file
+# OpenTracing JAX-RS Instrumentation
+
+Warning: This library is still a work in progress!
+
+[![Travis](https://travis-ci.org/opentracing-contrib/java-jaxrs.svg?branch=master)](https://travis-ci.org/opentracing-contrib/java-jaxrs)
+
+OpenTracing instrumentation for JAX-RS standard. 
+It supports server and client request tracing.
+
+## Tracing Server Requests
+```
+// register this in javax.ws.rs.core.Application
+ServerTracingDynamicFeature.Builder
+    .traceAll(yourPreferredTracer)
+    .withStandardTags()
+    .withDecorator(ServerSpanDecorator.HTTP_PATH_OPERATION_NAME)
+    .build();
+
+@GET
+@Path("/hello")
+@Traced(operationName = "helloRenamed") // optional, by default operation name is provided by ServerSpanDecorator
+public Response hello(@BeanParam CurrentSpan currentSpan) {
+    /**
+     * Some business logic
+    /*
+    final Span span = currentSpan.get();
+    Span childSpan = tracer.buildSpan("businessOperation")
+            .asChildOf(span)
+            .start())
+    childSpan.finish();
+
+    return Response.status(Response.Status.OK).build();
+}
+```
+
+## Tracing Client Requests
+```
+ClientTracingFeature.Builder
+    .traceAll(yourPreferredTracer, jaxRsClient)
+    .withStandardTags()
+    .withDecorator(ServerSpanDecorator.HTTP_PATH_OPERATION_NAME)
+    .build();
+
+Response response = jaxRsClient.target("http://localhost/endpoint")
+    .request()
+    .property(TracingProperties.CHILD_OF, parentSpan) // optional, by default new trace is started
```

Comment:
This is a deal breaker for me personally. Java as a language has a pretty good story for implicit context propagation via thread locals (e.g. via WIP https://github.com/opentracing-contrib/java-activespan), so requiring the users to remember to add code for tracing to every call site is a step back.